R/data-process.R

```diff
@@ -1,229 +1,273 @@
+
+  data_weighted_burden_wide_collapse <- data_weighted_burden_wide |>
+    dplyr::group_by(data_weighted_burden_wide$time) |>
+    dplyr::summarise(data_weighted_burden_wide$diff_daly_mean <- mean(data_weighted_burden_wide$diff_daly),
+                     data_weighted_burden_wide$diff_daly_min <- min(data_weighted_burden_wide$diff_daly),
+                     data_weighted_burden_wide$diff_daly_max <- max(data_weighted_burden_wide$diff_daly),
+                     data_weighted_burden_wide$diff_yll_mean <- mean(data_weighted_burden_wide$diff_yll),
+                     data_weighted_burden_wide$diff_yll_min <- min(data_weighted_burden_wide$diff_yll),
+                     data_weighted_burden_wide$diff_yll_max <- max(data_weighted_burden_wide$diff_yll),
+                     data_weighted_burden_wide$diff_yld_mean <- mean(data_weighted_burden_wide$diff_yld),
+                     data_weighted_burden_wide$diff_yld_min <- min(data_weighted_burden_wide$diff_yld),
+                     data_weighted_burden_wide$diff_yld_max <- max(data_weighted_burden_wide$diff_yld),
+                     data_weighted_burden_wide$cumdiff_daly_mean <- mean(data_weighted_burden_wide$cumdiff_daly),
+                     data_weighted_burden_wide$cumdiff_daly_min <- min(data_weighted_burden_wide$cumdiff_daly),
+                     data_weighted_burden_wide$cumdiff_daly_max <- max(data_weighted_burden_wide$cumdiff_daly),
+                     data_weighted_burden_wide$cumdiff_yll_mean <- mean(data_weighted_burden_wide$cumdiff_yll),
+                     data_weighted_burden_wide$cumdiff_yll_min <- min(data_weighted_burden_wide$cumdiff_yll),
+                     data_weighted_burden_wide$cumdiff_yll_max <- max(data_weighted_burden_wide$cumdiff_yll),
+                     data_weighted_burden_wide$cumdiff_yld_mean <- mean(data_weighted_burden_wide$cumdiff_yld),
+                     data_weighted_burden_wide$cumdiff_yld_min <- min(data_weighted_burden_wide$cumdiff_yld),
+                     data_weighted_burden_wide$cumdiff_yld_max <- max(data_weighted_burden_wide$cumdiff_yld))
+
+  return(data_weighted_burden_wide_collapse)
+}
+
+#' Perform data smoothing
+#'
+#' This function performs data smoothing.
+#'
+#' @param data_weighted_burden A data frame containing weighted values for burden of disease.
+#' @return A data frame with spline smoothing applied for burden of disease.
+#' @export
+gen_data_weighted_burden_spline <- function(data_weighted_burden) {
+
+  ## Only keep those 0 or negative values
+
+  ## Notes: Delete years 27,30,32-33 for ps3-low; Delete years 2028 for ps4-low
+
+  data_weighted_burden_mean <- data_weighted_burden |>
+    dplyr::filter(data_weighted_burden$cumdiff_daly_mean <= 0)
+
+  data_weighted_burden_min <- data_weighted_burden |>
+    dplyr::filter(data_weighted_burden$cumdiff_daly_min <= 0)
+
+  ## Notes: Delete years 29, 31 for ps2-high; Delete 37-38 for ps3-low; Delete 33-34 for ps4-middle; Delete 36-38 for ps4-low
```

Comment:
Also amend to "Notes for India project: Delete years 29, 31 for ps2-high; Delete 37-38 for ps3-low; Delete 33-34 for ps4-middle; Delete 36-38 for ps4-low"

Reply:
This also